test/test_helper.exs

```diff
@@ -2,6 +2,9 @@
 # Needs to be called before ExUnit.start
 {:ok, _} = Application.ensure_all_started(:ex_machina)
 
+# By default, exclude S3 tests
+# To run S3 tests, run `mix test --include s3`
+ExUnit.configure exclude: [:s3]
```

Comment:
So the reason I made this excluded by default is that these tests should be in the minority. The majority of the time, we won't be interacting with tests involving S3 / other secret env variables. Since it's likely most people won't mess with those tests, I think it makes sense to exclude by default and call it out in the README.

Also, as long as CI is set up to include those tests, we'll be sure to catch anything before it gets merged.

If we do want to include these by default, just let me know and I'll change it. Just wanted to give my two cents :)